pages/apis/graphql/cookbooks/pipeline_templates.md

```diff
@@ -0,0 +1,145 @@
+# Pipeline templates
+
+A collection of common tasks with pipeline templates using the GraphQL API.
+
+You can test out the Buildkite GraphQL API using the [Buildkite explorer](https://graphql.buildkite.com/explorer). This includes built-in documentation under the _Docs_ panel.
+
+## List pipeline templates
+
+Get the first 10 pipeline templates and their information for an organization:
+
+```graphql
+query GetPipelineTemplates {
+  organization(slug: "organization-slug") {
+    pipelineTemplates(first: 10) {
+      edges {
+        node {
+          id
+          uuid
+          name
+          description
+          configuration
+          available
+        }
+      }
+    }
+  }
+}
+```
+
+## Get a pipeline template
+
+Get information on a pipeline template, specifying the pipeline templates' UUID as the `uuid` argument of the `pipelineTemplate` query:
+
+```graphql
+query GetPipelineTemplate {
+  pipelineTemplate(uuid: "pipeline-template-uuid") {
+    id
+    uuid
+    name
+    description
+    configuration
+    available
+  }
+}
+```
+
+## Create a pipeline template
+
+Create a pipeline template for an organization using the `pipelineTemplateCreate` mutation:
+
+```graphql
+mutation CreatePipelineTemplate {
+  pipelineTemplateCreate(input: {
+    organizationId: "organization-id",
+    name: "template name",
+    description: "it does a thing",
+    configuration: "steps:\n  - command: deploy.sh",
+    available: false
+  }) {
+    pipelineTemplate {
+      id
+      uuid
+      name
+      description
+      configuration
+      available
+    }
+  }
+}
+```
+
+## Updating a pipeline template
+
+Update a pipeline template on an organization using the `pipelineTemplateUpdate` mutation, specifying the ID for organization and pipeline template:
+
+```graphql
+mutation UpdatePipelineTemplate {
+  pipelineTemplateUpdate(input: {
+    organizationId: "organization-id",
+    id: "pipeline-template-id",
+    configuration: "steps:\n - comand: updated_steps.sh"
+    available: true
+  }) {
+    pipelineTemplate {
+      id
+      uuid
+      name
+      description
+      configuration
+      available
+    }
+  }
+}
+```
+
+## Deleting a pipeline template
+
+Delete a pipeline template using the `pipelineTemplateDelete` mutation, specifying the ID for organization and pipeline template:
+
+```graphql
+mutation DeletePipelineTemplate {
+  pipelineTemplateDelete(input: {
+    organizationId: "organization-id",
+    id: "pipeline-template-id"
+  }) {
+    deletedPipelineTemplateId
+  }
+}
+```
+
+## Managing pipeline template assignment on a pipeline
+
+Admins and users with manage pipeline permissions can assign a pipeline template to a pipeline using the `pipelineUpdate` mutation:
```

Comment:
Is "manage pipeline" the name of the permission in the UI? If so, I think it would be good to use italics:

```suggestion
Admins and users with _manage pipeline permissions_ can assign a pipeline template to a pipeline using the `pipelineUpdate` mutation:
```

Or, consider the following alternate, which I think reads a little better:

```suggestion
Admins and users with permission to manage pipelines can assign a pipeline template to a pipeline using the `pipelineUpdate` mutation:
```

Reply:
"Manage pipeline" isn't the permission we use in the UI, I used it to collectively describe `read_project` and `create_project` permissions :).

Do you think we should be more specific with the permissions?

Reply:
Ah okay. Hmmm is there anywhere else that tells them that specific info? The first place I checked was our GraphQL schema docs, but that doesn't include permission requirements. The next place I looked was the Pipeline templates page, which has:

> anyone with permission to create or change a pipeline can assign a template.

Is that the equivalent info?

If so, I think it's fine to just use my second suggestion: "Admins and users with permission to manage pipelines can..."

Reply:
The closest page I can think of that has a similar granularity on permissions would be on the pipeline templates rest api page. Each of the endpoints list out what permissions and api token scopes are required in order to use the api.

Cool, I ran with your second suggestion in lieu of any other updates we may want to make in the future :)